### DISCUSSION TITLE
Is there a way to change the URL to a custom friendly url?

### DISCUSSION THREAD

**Discussion by dxburge** [mediacms-io/mediacms#1066] [Q&A]
Example would be instead of:
https://www.my-mediacrm-website.com/view?m=Vo2DJpTJm
to rewrite it to:
https://www.my-mediacrm-website.com/view?m=my_seo_friendly_video_name

**Comment by mgogoulos** [comment]
Sure, there is a unique token automatically created when a new video (or other media) is uploaded. You can then change that to what you want, but you have to do it through the admin page. Go to admin --> Files --> Media, find the media you want, edit it and set the `Friendly token:` to what you want. 

There's an alternative too, if you don't want to do it through the admin. You can edit files/forms.py and add the `friendly_token` field to the list of fields on MediaForm. If you do this, all users will be able to set it which is probably not a good idea, so you can make it only for admins or editors by removing it in case the user is not one of them. A few lines below, 

```
        if not is_mediacms_editor(user):
            self.fields.pop("featured")
```

add friendly_token there too

```
        if not is_mediacms_editor(user):
            self.fields.pop("featured")
            self.fields.pop("friendly_token")

```

Restart mediacms and you should see it

**Reply by touni1** [reply]
How can I make this automatic for each video that is uploaded, instead of having to edit each video individually?
